src/lib/stepper/stepper-horizontal.html

```diff
@@ -8,22 +8,14 @@
          [tabIndex]="_focusIndex == i ? 0 : -1"
          (click)="step.select()"
          (keydown)="_onKeydown($event)">
-      <div class="mat-stepper-index">
-        {{i + 1}}
-      </div>
-
-      <div class="mat-stepper-label">
-        <!-- If there is a label template, use it. -->
-        <ng-container *ngIf="step.stepLabel" [ngTemplateOutlet]="step.stepLabel.template">
-        </ng-container>
-        <!-- It there is no label template, fall back to the text label. -->
-        <div *ngIf="!step.stepLabel">{{step.label}}</div>
-      </div>
+      <md-step-icon [step]="step" [selected]="selectedIndex == i" [index]="i"></md-step-icon>
```

Comment:
could we not just combine these 2 internal components (and their parent div) into a single `md-step-header` component?